### PR TITLE
Update 3DS v2 docs

### PR DESCRIPTION
**Pull request by nuxzero**
## Purpose

This PR purposes to write documentation for 3DS v2.

## Description

- Create `docs/3d-secure-v2.md` that contains about how to implement the 3DS v2 in the merchant application.
- Write document to the source code.

## Quality assurance

N/A

## Operations impact

N/A

## Pre- and post-deployment steps

N/A

## Business impact (release notes)

N/A

## Add labels and assign reviewers

N/A

## Back-out Procedure

N/A

### CODE REVIEW

**Review by nikolack — APPROVED** [review]

**Review by igroomgrim — APPROVED** [review]

File: docs/3d-secure-v2.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# 3D Secure V2
+
+Unlike previous version the cardholder is redirected to the bank website, the 3D Secure V2 provided stronge customer authentication and the customer can authenticate within the merchant application.
+
+## Get Started
+
+To use the 3D Secure v2, the `AuthorizingPaymentConfig.initialize(config)` function must be called before starting the authorizing payment with the `authorize_uri` [see more](https://www.omise.co/how-to-implement-3-D-Secure).
```
Comment (danfowler) [inline docs/3d-secure-v2.md:7]:
```suggestion
To use 3DS2, the `AuthorizingPaymentConfig.initialize(config)` function must be called before starting the authorizing payment process with `authorize_uri` [see more](https://www.omise.co/how-to-implement-3-D-Secure).
```

File: docs/3d-secure-v2.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+          .uiCustomization(uiCustomization)
+          .timeout(5)
+          .build()
+  val authPaymentConfig = AuthorizingPaymentConfig.Builder()
+          .threeDSConfig(threeDSConfig)
+          .build()
+  AuthorizingPaymentConfig.initialize(authPaymentConfig)
+```
+
+These are available configuration in the 3DS v2.
```
Comment (danfowler) [inline docs/3d-secure-v2.md:20]:
```suggestion
These are the available configurations in 3DS2.
```

File: docs/3d-secure-v2.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+To use the authentication page, the Omise SDK provides the `AuthorizingPaymentActivity` for handle the authorizing payment. You can declare the `AuthorizingPaymentActivity` in your `AndroidManifest.xml` file.
+
+```xml
+<activity
+  android:name="co.omise.android.ui.AuthorizingPaymentActivity"
+  android:theme="@style/OmiseTheme" />
+```
+
+In your activity, you can use the `startActivityForResult()` function to start the `AuthorizingPaymentActivity` and handle result.
```
Comment (danfowler) [inline docs/3d-secure-v2.md:35]:
```suggestion
In your activity, you can use the `startActivityForResult()` function to start the `AuthorizingPaymentActivity` and handle the result.
```

File: docs/3d-secure-v2.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+```kotlin
+val intent = Intent(this, AuthorizingPaymentActivity::class.java)
+intent.putExtra(AuthorizingPaymentURLVerifier.EXTRA_AUTHORIZED_URLSTRING, AUTHORIZED_URL)
+intent.putExtra(AuthorizingPaymentURLVerifier.EXTRA_EXPECTED_RETURN_URLSTRING_PATTERNS, EXPECTED_URL_PATTERNS)
+intent.putExtra(OmiseActivity.EXTRA_PKEY, PUBLIC_KEY)
+intent.putExtra(OmiseActivity.EXTRA_TOKEN, TOKEN_ID)
+startActivityForResult(intent, AUTHORIZING_PAYMENT_REQUEST_CODE)
+```
+
+After the cardholder had completed the authorizing payment process,  the `onActivityResult()` function will be called, you can handle the authorizing payment result in there.
```
Comment (danfowler) [inline docs/3d-secure-v2.md:46]:
```suggestion
After the cardholder completes the authorizing payment process,  the `onActivityResult()` function will be called, you can handle the authorizing payment result there.
```

File: docs/3d-secure-v2.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        // Available only in 3D Secure V1
+        val url = data?.getStringExtra(AuthorizingPaymentURLVerifier.EXTRA_RETURNED_URLSTRING)
+        // Available only in 3D Secure V2
+        val token = data?.getParcelableExtra<Token>(OmiseActivity.EXTRA_TOKEN_OBJECT)
+    }
+}
+```
+
+
+You can checkout the sample implementation in the [PaymentProcessingActivity](sample-threeds-v2-app/src/main/java/co/omise/android/example/PaymentProcessingActivity.kt) class in the **sample-threeds-v2-app** app.
```
Comment (danfowler) [inline docs/3d-secure-v2.md:61]:
```suggestion
You can check out the sample implementation in the [PaymentProcessingActivity](sample-threeds-v2-app/src/main/java/co/omise/android/example/PaymentProcessingActivity.kt) class in the **sample-threeds-v2-app** app. 
```

File: docs/3d-secure-v2.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+                  .textFontName("RobotoMono-Regular.ttf")
+                  .textFontColor("#000000")
+                  .textFontSize(16)
+                  .build())
+  val threeDSConfig = ThreeDSConfig.Builder()
+          .uiCustomization(uiCustomization)
+          .build()
+```
+
+You can checkout the [UiCustomization](/sdk/src/main/java/co/omise/android/config/UiCustomization.kt) class to see customizable elements that you can customize in the challenge flow.
```
Comment (danfowler) [inline docs/3d-secure-v2.md:78]:
```suggestion
You can check out the [UiCustomization](/sdk/src/main/java/co/omise/android/config/UiCustomization.kt) class to see customizable elements that you can customize in the challenge flow.
```

File: sdk/src/main/java/co/omise/android/config/ThreeDSConfig.kt [hunk 7]
```diff
@@ -2,32 +2,58 @@ package co.omise.android.config
 
 import androidx.annotation.IntRange
 
+/**
+ * Configuration for 3DS V2.
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/config/ThreeDSConfig.kt:6]:
```suggestion
 * Configuration for 3DS2
```

File: docs/3d-secure-v2.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
+# 3D Secure V2
```
[the remark this answers is not in the document]
Comment (danfowler) [reply docs/3d-secure-v2.md:1]: I recommend using "3D Secure 2" and "3DS2" as a shorthand.

**Review by danfowler — APPROVED** [review]
We can check again once you've committed the changes (it's hard to edit well in GitHub 😄)

**Comment by nuxzero** [comment]
@danfowler I made some changes from your suggestions.

File: docs/3d-secure-v2.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
+# 3D Secure 2
+
+The previous version of 3DS redirected the cardholder to the bank's website,
```
Comment (danfowler) [inline docs/3d-secure-v2.md:3]:
```suggestion
The previous version of 3D Secure (3DS) redirected the cardholder to the bank's website,
```

**Review by danfowler — APPROVED** [review]

**Comment by danfowler** [comment]
@nuxzero will you link from the main README.md?

**Comment by nuxzero** [comment]
@danfowler Let me add it. Thanks.

**Comment by nuxzero** [comment]
@danfowler I just added the link and comments to the sample code.

**Pull request merged**